apps/ledger-live-desktop/src/renderer/components/SignMessageConfirm/index.tsx

```diff
@@ -12,29 +12,50 @@ import useTheme from "~/renderer/hooks/useTheme";
 import Text from "~/renderer/components/Text";
 import Box from "~/renderer/components/Box";
 import { getLLDCoinFamily } from "~/renderer/families";
+import FormattedVal from "~/renderer/components/FormattedVal";
 
 const FieldText = styled(Text).attrs(() => ({
   ml: 1,
   ff: "Inter|Medium",
   color: "palette.text.shade80",
   fontSize: 3,
-}))`
-  word-break: break-all;
-  text-align: right;
-  max-width: 50%;
-`;
+}))``;
 
 export type FieldComponentProps = {
-  account: AccountLike;
+  account: Account;
   field: DeviceTransactionField;
+  contractAddress?: string;
 };
 
 export type FieldComponent = React.ComponentType<FieldComponentProps>;
 
-const TextField = ({ field }: FieldComponentProps) => {
+const TextField = ({ field, account, contractAddress }: FieldComponentProps) => {
+  const isPropertyAmount = field.label.includes("Amount") && contractAddress;
+
+  const unit = isPropertyAmount
+    ? account.subAccounts?.find(a => a.token.contractAddress === contractAddress)?.token.units[0]
+    : undefined;
+
   return field.type === "text" ? (
     <SignMessageConfirmField label={field.label}>
-      <FieldText>{field.value}</FieldText>
+      <FieldText>
```

Comment:
I don't think I ever saw a `FormattedVal` in a `FieldText`, I think we usually just use it as a `Text` component, is there any benefit here to nest it in the `FieldText` as it should already be using the same font